include/mbgl/util/string.hpp

```diff
@@ -10,7 +10,16 @@ namespace util {
 
 template <class T>
 inline std::string toString(T t) {
-    return std::to_string(t);
+    if (t - (int)t > 0.0 || t + (int)t < 0.0) {
```

Comment:
`std::to_string()` can also be used with numeric types much larger than `int`, as well as unsigned types. Casting an `unsigned` or `long double` to an `int` risks overflow and other unexpected behavior.